File: JAVA/PedraPapelAtaqueAereo.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.Scanner;

public class PedraPapelAtaqueAereo {

    private final static String ATAQUE = "ataque";
    private final static String PAPEL = "papel";
    private final static String PEDRA = "pedra";
    private final static String P1_WIN = "Jogador 1 venceu";
    private final static String P2_WIN = "Jogador 2 venceu";
    private final static String BOTH_WIN = "Ambos venceram";
    private final static String NO_WINNER = "Sem ganhador";
    private final static String ANIQUILATION = "Aniquilacao mutua";

    public static void main(String[] args) {
		Scanner sc = new Scanner( System.in );
        int N = sc.nextInt();

        for ( int i=0; i < N; i++ ) {
            String p1 = sc.next();
            String p2 = sc.next();

            System.out.println( jokenpoWinner( p1, p2 ) );
        }
        sc.close();
	}

    /**
     * Check who is the winner of the game Jokenpo.
     */
    public static String jokenpoWinner(String p1, String p2) {
        String result = P2_WIN;

        if ( p1.equalsIgnoreCase( ATAQUE ) && p2.equalsIgnoreCase( ATAQUE ) ) {
            result = ANIQUILATION;
        } else if ( p1.equalsIgnoreCase( PEDRA ) && p2.equalsIgnoreCase( PEDRA ) ) {
            result = NO_WINNER;
        } else if ( p1.equalsIgnoreCase( PAPEL ) && p2.equalsIgnoreCase( PAPEL ) ) {
            result = BOTH_WIN;
        } else if ( p1.equalsIgnoreCase( ATAQUE ) && p2.equalsIgnoreCase( PEDRA ) || p2.equalsIgnoreCase( PAPEL ) ) {
            result = P1_WIN;
        } else if ( p1.equalsIgnoreCase( PEDRA ) && p2.equalsIgnoreCase( PAPEL ) ) {
            result = P1_WIN;
        }

        return result;
    }
}

File: GO/PedraPapelAtaqueAereo.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "strings"

const ATAQUE = "ataque"
const PAPEL = "papel"
const PEDRA = "pedra"
const P1_WIN = "Jogador 1 venceu"
const P2_WIN = "Jogador 2 venceu"
const BOTH_WIN = "Ambos venceram"
const NO_WINNER = "Sem ganhador"
const ANIQUILATION = "Aniquilacao mutua"

/**
 * Check who is the winner of the game Jokenpo.
 */
func jokenpoWinner(p1 string, p2 string) string {
	result := P2_WIN
	p1 = strings.ToLower(p1)
	p2 = strings.ToLower(p2)

	if p1 == ATAQUE && p2 == ATAQUE {
		result = ANIQUILATION
	} else if p1 == PEDRA && p2 == PEDRA {
		result = NO_WINNER
	} else if p1 == PAPEL && p2 == PAPEL {
		result = BOTH_WIN
	} else if p1 == ATAQUE && p2 == PEDRA || p2 == PAPEL {
		result = P1_WIN
	} else if p1 == PEDRA && p2 == PAPEL {
		result = P1_WIN
	}

	return result
}
